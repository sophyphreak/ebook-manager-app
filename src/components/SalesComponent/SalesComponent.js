import React from 'react';
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  FormGroup,
  Input,
  Label,
  Row
} from 'reactstrap';

import BasicInput from '../FormElements/MainElements/BasicInput';

const SalesComponent = ({
  bsr,
  sales,
  onBsrChange
}) => (
  <Row className="animated fadeIn">
    <Col xs="12" sm={{ size: 6, offset: 3 }}>
      <Card>
        <CardHeader>
          <h4><strong>Sales Estimator</strong></h4>
        </CardHeader>
        <CardBody>
          <BasicInput
            label="Enter BSR"
            onChange={onBsrChange}
            value={bsr}
          />
          <br />
          {
            bsr &&
            <FormGroup className="animated fadeIn">
              <Label>Estimated sales</Label>
              <br /> 
              <Label><b>{sales} units per month</b></Label>
            </FormGroup>
          }
        </CardBody>
      </Card>
    </Col>
  </Row>
);

export default SalesComponent;