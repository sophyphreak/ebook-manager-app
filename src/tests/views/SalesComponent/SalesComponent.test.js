import React from 'react';
import { shallow } from 'enzyme';
import SalesComponent from '../../../views/SalesComponent/SalesComponent';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<SalesComponent />);
});

test('should render SalesComponent correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
