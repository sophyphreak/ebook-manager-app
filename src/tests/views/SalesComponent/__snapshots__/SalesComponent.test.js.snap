// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SalesComponent correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SalesComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Col
        sm={
          Object {
            "offset": 3,
            "size": 6,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Card
          tag="div"
        >
          <CardHeader
            tag="div"
          >
            <h4>
              <strong>
                Sales Estimator
              </strong>
            </h4>
          </CardHeader>
          <CardBody
            tag="div"
          >
            <KindleOrPrint />
            <BasicInput
              label="Enter BSR"
            />
            <br />
          </CardBody>
        </Card>
      </Col>,
      "className": "animated fadeIn",
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Card
          tag="div"
        >
          <CardHeader
            tag="div"
          >
            <h4>
              <strong>
                Sales Estimator
              </strong>
            </h4>
          </CardHeader>
          <CardBody
            tag="div"
          >
            <KindleOrPrint />
            <BasicInput
              label="Enter BSR"
            />
            <br />
          </CardBody>
        </Card>,
        "sm": Object {
          "offset": 3,
          "size": 6,
        },
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "xs": "12",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader
              tag="div"
            >
              <h4>
                <strong>
                  Sales Estimator
                </strong>
              </h4>
            </CardHeader>,
            <CardBody
              tag="div"
            >
              <KindleOrPrint />
              <BasicInput
                label="Enter BSR"
              />
              <br />
            </CardBody>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h4>
                <strong>
                  Sales Estimator
                </strong>
              </h4>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Sales Estimator
                </strong>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sales Estimator",
                },
                "ref": null,
                "rendered": "Sales Estimator",
                "type": "strong",
              },
              "type": "h4",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <KindleOrPrint />,
                <BasicInput
                  label="Enter BSR"
                />,
                <br />,
                undefined,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "kindleOrPrint": undefined,
                  "onKindleOrPrintChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Enter BSR",
                  "onChange": undefined,
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              undefined,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          sm={
            Object {
              "offset": 3,
              "size": 6,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Card
            tag="div"
          >
            <CardHeader
              tag="div"
            >
              <h4>
                <strong>
                  Sales Estimator
                </strong>
              </h4>
            </CardHeader>
            <CardBody
              tag="div"
            >
              <KindleOrPrint />
              <BasicInput
                label="Enter BSR"
              />
              <br />
            </CardBody>
          </Card>
        </Col>,
        "className": "animated fadeIn",
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Card
            tag="div"
          >
            <CardHeader
              tag="div"
            >
              <h4>
                <strong>
                  Sales Estimator
                </strong>
              </h4>
            </CardHeader>
            <CardBody
              tag="div"
            >
              <KindleOrPrint />
              <BasicInput
                label="Enter BSR"
              />
              <br />
            </CardBody>
          </Card>,
          "sm": Object {
            "offset": 3,
            "size": 6,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "xs": "12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeader
                tag="div"
              >
                <h4>
                  <strong>
                    Sales Estimator
                  </strong>
                </h4>
              </CardHeader>,
              <CardBody
                tag="div"
              >
                <KindleOrPrint />
                <BasicInput
                  label="Enter BSR"
                />
                <br />
              </CardBody>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h4>
                  <strong>
                    Sales Estimator
                  </strong>
                </h4>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    Sales Estimator
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sales Estimator",
                  },
                  "ref": null,
                  "rendered": "Sales Estimator",
                  "type": "strong",
                },
                "type": "h4",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <KindleOrPrint />,
                  <BasicInput
                    label="Enter BSR"
                  />,
                  <br />,
                  undefined,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "kindleOrPrint": undefined,
                    "onKindleOrPrintChange": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Enter BSR",
                    "onChange": undefined,
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                undefined,
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
